fluent/client/ws/connection.go

```diff
@@ -279,24 +270,21 @@ func (wsc *connection) runReadLoop(nextMsg chan connMsg) {
 		msg.mt, msg.message, msg.err = wsc.Conn.ReadMessage()
 
 		if msg.err != nil {
-			wsc.logger.Println("error received: ", msg.err.Error())
-		}
-
-		if wsc.hasConnState(ConnStateClosed) && errors.Is(msg.err, net.ErrClosed) {
-			// "healthy" close
-			wsc.logger.Println("connection closed and terminated")
+			if wsc.hasConnState(ConnStateClosed) && errors.Is(msg.err, net.ErrClosed) {
+				// healthy close
+				break
+			}
 
-			break
+			var err net.Error
+			if errors.As(msg.err, &err) || errors.Is(msg.err, net.ErrClosed) ||
+				websocket.IsCloseError(msg.err, websocket.CloseAbnormalClosure) {
+				wsc.setConnState(ConnStateError)
```

Comment:
Setting ConnStateError prevents Close from trying to send close message to peer.

https://github.com/IBM/fluent-forward-go/pull/31/files#diff-f4b3bc6364cb918edaf959e398736dddfa88d9ee58ed9c9c094c00d11d429e6eR195